src/listeners/ticket/close-modal-submit-validator.ts

```diff
@@ -0,0 +1,120 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { Events, Listener } from '@sapphire/framework';
+import {
+    ChannelType,
+    EmbedBuilder,
+    GuildMember,
+    type Interaction,
+    type ModalSubmitInteraction,
+} from 'discord.js';
+import { Color } from '../../lib/embeds';
+import { InputId, ModalId } from '../../lib/ticket';
+import { messageLogger } from '../..';
+import { userUrl } from '../../lib/user';
+import { ticketLogs } from '../..';
+
+@ApplyOptions<Listener.Options>({ event: Events.InteractionCreate })
+export class InteractionCreateListener extends Listener<
+    typeof Events.InteractionCreate
+> {
+
+    public override async run(interaction: Interaction) {
+        if (
+            !interaction.isModalSubmit() ||
+            interaction.customId !== ModalId.Close ||
+            !interaction.inGuild()
+        ) {
+            return;
+        }
+
+        await interaction.deferReply({ ephemeral: true });
+
+        const resolution = interaction.fields.getTextInputValue(InputId.Resolution).trim();
+        if (!resolution) {
+            return this.sendValidationFailure(
+                interaction,
+                'Resolution must contain at least 1 non-whitespace character'
+            );
+        }
+
+        const guild = await interaction.client.guilds.fetch(interaction.guildId);
+        const member = await guild.members.fetch(interaction.member.user.id);
+
+        const ticketThreadId = interaction.channelId;
+        if (!ticketThreadId) {
+            return;
+        }
+        const ticketThread = await guild.channels.fetch(
```

Comment:
Can fetch the thread directly from the client's channel manager, rather than having to use `guild`'s (meaning the fetch of `guild` can move even further down).
```suggestion
        const ticketThread = await interaction.client.channels.fetch(
```